File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Square.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Square implements Shape {
    public double sideLen;

    public Square(double sideLen) {
        this.sideLen = sideLen;
    }
    
    public double getArea() {
        return sideLen * sideLen;
    }

    public double getPerimeter() {
        return 4 * sideLen;
    }
}

File: typescript/src/shapes/square.ts
import { Shape } from "./shape"

function square(sideLen: number): Shape {
    return {
        computeArea: function (): number {
            return sideLen * sideLen
        },
        computePerimeter: function (): number {
            return 4 * sideLen
        }
    }
}

export { square }